src/entry.rs

```diff
@@ -639,6 +628,31 @@ impl<'a> EntryFields<'a> {
         }
         return Ok(Unpacked::File(f));
 
+        #[cfg_attr(target_os = "wasi", allow(unused_variables))]
+        fn set_mtime(dst: &Path, f: &mut std::fs::File, mtime: u64) -> Result<(), TarError> {
+            if cfg!(target_os = "wasi") {
+                // The filetime crate does not yet implement updating file times but hopefully
+                // the upstream developers will incorporate it at which time this can be
+                // implemented in this crate as well.
+                // see: https://github.com/alexcrichton/filetime/blob/master/src/wasm.rs#L34
+                return Err(TarError::new(format!("failed to set mtime for `{}`", dst.display()),
+                    io::Error::new(io::ErrorKind::Other, "setting modified date/times is not yet supported on WASI - omit preserve_mtime to avoid this error")));
+            }
```

Comment:
Does this need to be specifically handled for wasi? Could the current error be deferred to instead of explicitly returning an error?

Reply:
Sure no problem - checked the code and it will work fine like that so removed this call branch